### ISSUE TITLE
Limit WebP image dimensions

### ISSUE THREAD

**Issue by adamdunnage** [WordPress/performance#588]
## Feature Description

As mentioned by a [user in the support forum](https://wordpress.org/support/topic/limiting-image-dimensions/), would it be possible to add a filter or option to allow users to limit WebP image dimensions.  For example: limiting any WebP images created to 1024x1024px. 

We do have a similar filter that can limit the number image sizes to generate webp images at:

```
add_filter(
	'webp_uploads_image_sizes_with_additional_mime_type_support',
	function( $sizes ) {
		$sizes['allowed_size_400x300'] = true;
		return $sizes;
	}
);
```

So would it be possible to add something similar for the image dimensions?

**Comment by adamsilverstein** [comment]
>  limit WebP image dimensions. For example: limiting any WebP images created to 1024x1024px.

@adamdunnage Would this be used in the mode where both JPEG and WebP are generated? I will check if this is already possible with the current filters.

**Comment by adamdunnage** [comment]
@adamsilverstein Yes, if that is possible then that would be helpful I believe. Thanks for checking!

**Comment by the-hercules** [comment]
Hi @adamdunnage,

I’d like to work on this and just need a quick clarification before proceeding.

### Question

Would you prefer this as:
- A developer-only filter (similar to `webp_uploads_image_sizes_with_additional_mime_type_support`), or  
- A GUI setting under **Settings → Media**?

### Additional Context

While testing, I noticed that large images often don’t generate AVIF/WebP even when both formats are enabled. This appears to be due to the existing `webp_uploads_discard_larger_generated_images`,`webp_uploads_should_discard_additional_image_file` filter, which discards the modern format if it ends up larger than the JPEG.

Given that:
- If “JPEG + modern format” mode is disabled, we would still need scaled JPEG images anyway.
- In dual mode, modern formats are already discarded when larger than the JPEG.

Should the proposed dimension limit strictly prevent generation above a certain width/height regardless of filesize comparison, or should it work alongside the existing discard logic?
cc- @adamsilverstein

**Comment by westonruter** [comment]
I'm not sure we need this. WordPress already has a [`big_image_size_threshold`](https://developer.wordpress.org/reference/hooks/big_image_size_threshold/) filter:

> If the original image width or height is above the threshold, it will be scaled down. The threshold is used as max width and max height. The scaled down image will be used as the largest available size, including the _wp_attached_file post meta value.

**Comment by adamsilverstein** [comment]
Closing this as "works as expected", since as @westonruter pointed out WordPress already has a [big_image_size_threshold](https://developer.wordpress.org/reference/hooks/big_image_size_threshold/) filter that I believe addresses this issue. If I misses something @adamdunnage feel free to re-open.

**Comment by nickchomey** [comment]
I dont think that `big_image_size_threshold` does what is being asked here. It only controls the `-scaled` image size, which "replaces" the original image, but does nothing for the subsizes. Someone could, in theory, register an image size of 10000px and this would be generated irrespective of the  `big_image_size_threshold` filter value (if the original image is large enough)

(I've spent dozens of hours in the past week digging into this exact mechanism, to fix some longstanding trac tickets, and just further improve the performance).

Though, perhaps the `intermediate_image_sizes`  or `intermediate_image_sizes_advanced` filters already serve the requested purpose...? Likewise, its not clear to me what is being requested beyond the initially-cited `webp_uploads_image_sizes_with_additional_mime_type_support` filter

edit: Ah, i see. The [webp_uploads_get_image_sizes_additional_mime_type_support() ](https://github.com/WordPress/performance/blob/ea77bb13aabbc3aaf66ec280116a564db89e303b/plugins/webp-uploads/hooks.php#L706-L708) function just returns essentially booleans, stripping the size data from the sizes returned from `wp_get_additional_image_sizes()`. By the time it reaches that filter, the size data is gone
